Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad_Linux.cpp

```diff
@@ -0,0 +1,613 @@
+            { BTN_EAST_MASK,       &InputDeviceGamepad::Button::B },
+            { BTN_WEST_MASK,       &InputDeviceGamepad::Button::X },
+            { BTN_NORTH_MASK,      &InputDeviceGamepad::Button::Y }
+        };
+        return map;
+    }
+
+} // namespace
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace AzFramework
+{
+    using namespace GamepadLinuxPrivate;
+
+    InputDeviceGamepad::Implementation* InputDeviceGamepadLinux::Create(InputDeviceGamepad& inputDevice, AZStd::shared_ptr<AzFramework::LibEVDevWrapper> libevdevWrapper)
+    {
+        return aznew InputDeviceGamepadLinux(inputDevice, libevdevWrapper);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    InputDeviceGamepadLinux::InputDeviceGamepadLinux(InputDeviceGamepad& inputDevice, 
+    AZStd::shared_ptr<AzFramework::LibEVDevWrapper> libevdevWrapper)
+        : InputDeviceGamepad::Implementation(inputDevice)
+        , m_rawGamepadState(GetDigitalButtonIdByBitMaskMap())
+        , m_isConnected(false)
+    {
+        // These values are adjusted when a joystick is detected.  So these just exist
+        // to be placeholders until we actually compute them based on the data from the device.
+        m_rawGamepadState.m_triggerDeadZoneValue = 0.1f;
+        m_rawGamepadState.m_thumbStickLeftDeadZone = 0.1f;
+        m_rawGamepadState.m_thumbStickRightDeadZone = 0.1f;
+
+        // O3DE expects trigger values to always be 0.0 being not pressed at all, and 1.0 being pressed all the way.
+        // so far, devices tested have always had a range of 0 - 255, so this does not need special treatment like the axes do,
+        // and will get the normal maximum value straight from the device.
+        m_rawGamepadState.m_triggerMaximumValue = 1.0f;
+
+        // these values are always 1.0f, because we will normalize the value between -1.0 and 1.0.
+        // O3DE expects axis 0.0 to be the center but some devices have a range like 0 - 255 with 128 being the center.
+        // to account for that, this code will adjust and offset and scale.
+        m_rawGamepadState.m_thumbStickMaximumValue = 1.0f;
+
+        m_internalState.reset(new InternalState(libevdevWrapper));
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    InputDeviceGamepadLinux::~InputDeviceGamepadLinux()
+    {
+        m_internalState.reset();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    bool InputDeviceGamepadLinux::IsConnected() const
+    {
+        return m_isConnected;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceGamepadLinux::SetVibration(float leftMotorSpeedNormalized,
+                                                 float rightMotorSpeedNormalized)
```

Comment:
This might trigger an unused variable warning (as error)
```suggestion
    void InputDeviceGamepadLinux::SetVibration([[maybe_unused]] float leftMotorSpeedNormalized,
                                                 [[maybe_unused]] float rightMotorSpeedNormalized)
```